src/passes/OptimizeCasts.cpp

```diff
@@ -88,37 +92,43 @@ namespace wasm {
 
 namespace {
 
-// Find the best sources for local.gets: other locals with the same value, but
-// cast to a more refined type.
-struct BestSourceFinder : public LinearExecutionWalker<BestSourceFinder> {
+// Find the best casted verisons of local.gets: other locals with the same
+// value, but cast to a more refined type.
+struct BestCastFinder : public LinearExecutionWalker<BestCastFinder> {
 
   PassOptions options;
 
-  // A map of the best local.get for a particular index: the local.get that has
-  // the most refined type.
-  std::unordered_map<Index, Expression*> bestSourceForIndexMap;
-
-  // For each expression, a vector of local.gets that would like to use its
-  // value instead of themselves (as it is more refined).
-  std::unordered_map<Expression*, std::vector<LocalGet*>> requestMap;
-
-  static void doNoteNonLinear(BestSourceFinder* self, Expression** currp) {
-    self->bestSourceForIndexMap.clear();
+  // Map local indices to the most refined downcastings of local.gets from those
+  // indices.
+  //
+  // This is tracked in each basic block, and cleared between them.
+  std::unordered_map<Index, Expression*> mostCastedGets;
+
+  // For each most-downcasted local.get, a vector of other local.gets that could
+  // be replaced with gets of the downcasted value.
+  //
+  // This is tracked til the end of the entire function, and contains the
```

Comment:
```suggestion
  // This is tracked until the end of the entire function, and contains the
```

`'til` is an abbreviation of `until` (note the apostrophe), but not a word on its own.

Reply:
Thanks, good to know...